pkg/config/setup/otlp.go

```diff
@@ -12,86 +12,66 @@ import (
-	config.SetKnown(OTLPMetrics + ".*")
-	config.SetKnown(OTLPReceiverSection)
-	// Set all subkeys of otlp_config.receiver as known
-	config.SetKnown(OTLPReceiverSection + ".*")
-	config.SetKnown(OTLPDebug)
-	// Set all subkeys of otlp_config.debug as known
-	config.SetKnown(OTLPDebug + ".*")
-
-	// set environment variables for selected fields
-	setupOTLPEnvironmentVariables(config)
-}
-
-// setupOTLPEnvironmentVariables sets up the environment variables associated with different OTLP ingest settings:
-// If there are changes in the OTLP receiver configuration, they should be reflected here.
-//
-// We don't need to set the default value: it is dealt with at the unmarshaling level
-// since we get the configuration through GetStringMap
-//
-// We are missing TLS settings: since some of them need more work to work right they are not included here.
-func setupOTLPEnvironmentVariables(config pkgconfigmodel.Setup) {
-	// gRPC settings
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.endpoint")
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.transport")
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.max_recv_msg_size_mib")
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.max_concurrent_streams")
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.read_buffer_size")
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.write_buffer_size")
-	config.BindEnv(OTLPSection + ".receiver.protocols.grpc.include_metadata")
+	// Logs
+	config.BindEnvAndSetDefault("otlp_config.logs.enabled", false)
 
 	// Traces settings
+	config.BindEnvAndSetDefault("otlp_config.traces.enabled", true)
+	config.BindEnvAndSetDefault("otlp_config.traces.span_name_as_resource_name", false)
 	config.BindEnvAndSetDefault("otlp_config.traces.span_name_remappings", map[string]string{})
-	config.BindEnv("otlp_config.traces.span_name_as_resource_name")
-	config.BindEnvAndSetDefault("otlp_config.traces.probabilistic_sampler.sampling_percentage", 100.,
-		"DD_OTLP_CONFIG_TRACES_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE")
+	config.BindEnvAndSetDefault("otlp_config.traces.probabilistic_sampler.sampling_percentage", 100., "DD_OTLP_CONFIG_TRACES_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE")
+	config.BindEnvAndSetDefault("otlp_config.traces.internal_port", 5003)
+
+	// gRPC settings
+	config.BindEnv("otlp_config.receiver.protocols.grpc.endpoint")                              // TODO OTLP team: add default value
+	config.BindEnv("otlp_config.receiver.protocols.grpc.transport")                             // TODO OTLP team: add default value
+	config.BindEnv("otlp_config.receiver.protocols.grpc.max_recv_msg_size_mib")                 // TODO OTLP team: add default value
+	config.BindEnv("otlp_config.receiver.protocols.grpc.read_buffer_size")                      // TODO OTLP team: add default value
+	config.BindEnv("otlp_config.receiver.protocols.grpc.write_buffer_size")                     // TODO OTLP team: add default value
```

Comment:
We found an issue with Viper, where if we set a default value , it enables OTLP ingest. Is there a need for setting default values ?